dns-server/src/storage.rs

```diff
@@ -890,7 +896,7 @@ mod test {
         // Verify the initial configuration.
         assert!(generations_with_trees(&tc.store).is_empty());
         let config = tc.store.dns_config().await.unwrap();
-        assert_eq!(config.generation, 0);
+        assert_eq!(config.generation, Generation::from_u32(0));
```

Comment:
FWIW I found it pretty inconsistent when `.into()` would work and when it wouldn't, so I ended up switching to this midway.